makefile

```diff
@@ -31,7 +30,8 @@ include make/libraries
 # Set default compiler options.
 ##
 CFLAGS = -I . -isystem $(EIGEN) -isystem $(BOOST) -isystem$(CVODES)/include -Wall -DBOOST_RESULT_OF_USE_TR1 -DBOOST_NO_DECLTYPE -DBOOST_DISABLE_ASSERTS -DNO_FPRINTF_OUTPUT -pipe
-CFLAGS_GTEST = -DGTEST_USE_OWN_TR1_TUPLE
+CXXFLAGS = -std=c++14
```

Comment:
I think it's a step in the right direction towards better use of `make` (though maybe we want to get to `cmake` or something...?). `CXXFLAGS` will be used in implicit rules and the `COMPILE.cc` macro: http://docs.oracle.com/cd/E19504-01/802-5880/6i9k05dhg/index.html

Though you are correct that I don't think we make use of that indirection right now. I'm like halfway through making the make system in the math lib distinguish between C and C++ targets more properly, but I think it'd be best to do that in a separate pull request. I might then try to include more of the math library's makefiles in the Stan makefile to reduce duplication.